geozero/src/wkt/mod.rs

```diff
@@ -40,18 +61,45 @@ pub(crate) mod conversion {
 mod wkb {
     use crate::error::Result;
     use crate::wkb::{FromWkb, WkbDialect};
-    use crate::wkt::{WktString, WktWriter};
+    use crate::wkt::{EwktString, WktDialect, WktString, WktWriter};
+    use crate::CoordDimensions;
     use std::io::Read;
 
     impl FromWkb for WktString {
         fn from_wkb<R: Read>(rdr: &mut R, dialect: WkbDialect) -> Result<Self> {
             let mut out: Vec<u8> = Vec::new();
-            let mut writer = WktWriter::new(&mut out);
+            let mut writer = WktWriter::new(&mut out, WktDialect::Wkt);
             crate::wkb::process_wkb_type_geom(rdr, &mut writer, dialect)?;
             let wkt = String::from_utf8(out).map_err(|_| {
                 crate::error::GeozeroError::Geometry("Invalid UTF-8 encoding".to_string())
             })?;
             Ok(WktString(wkt))
         }
     }
+
+    impl FromWkb for EwktString {
+        fn from_wkb<R: Read>(rdr: &mut R, dialect: WkbDialect) -> Result<Self> {
+            let mut out: Vec<u8> = Vec::new();
+            let mut writer = WktWriter::new(&mut out, WktDialect::Ewkt);
+            writer.dims = CoordDimensions::xyzm();
+            crate::wkb::process_wkb_type_geom(rdr, &mut writer, dialect)?;
+            let wkt = String::from_utf8(out).map_err(|_| {
+                crate::error::GeozeroError::Geometry("Invalid UTF-8 encoding".to_string())
+            })?;
+            Ok(EwktString(wkt))
+        }
+    }
+}
+
+/// WKB dialect.
+#[derive(PartialEq, Debug, Clone, Copy)]
+pub enum WktDialect {
+    Wkt,
+    Ewkt,
+}
+
+impl Default for WktDialect {
+    fn default() -> Self {
+        WktDialect::Wkt
+    }
```

Comment:
```suggestion
#[derive(Default, PartialEq, Debug, Clone, Copy)]
pub enum WktDialect {
    #[default]
    Wkt,
    Ewkt,
```

Reply:
This is the first Rust project I'm contributing to and don't know all the niceties yet, thanks for showing me!